app/src/main/java/com/example/poste/PosteApplication.java

```diff
@@ -44,6 +99,38 @@ public static void setCurrentUser(User currentUser) {
         PosteApplication.currentUser = currentUser;
```

Comment:
This a static reference to the current user. It seems throughout the code this is how methods find the current user  Object and subsequently call the methods on it (like get folders). If the currentUser is a singleton, i don't think that changes how it should be referenced and located. 
Or the User class has also been made static?

Reply:
Take a look at the dev branch to get an idea of the implementation. User is a singleton, which means we can access it from anywhere, and always get the same User instance while ensuring only one User is ever created at runtime.

You can get the User instance via User.getUser(), and then call any methods on it as necessary.

The code was previously written in a different fashion; I thought that was part of what we were aiming to refactor, however. The 'new' style is what is currently being used in the new Login and Register activities

Reply:
Looking over the most resent dev.
Do we now use the static method: User.getUser(); anywhere that the code used PosteApplication.currentUser?

Reply:
That is the intention, I believe, but we are still refactoring.

Reply:
Also, by using that, we could set a currentFolder and currentPost on user (if you want to use that approach) rather than PosteApplication. That is a design choice however

Reply:
Im converting mine now to use this new method. but the old "currentUser" needs to get extracted out and the old data model java file need to be refactored out of the code. I will volunteer for this user story next sprint.

Reply:
Will remove in next user story. Only waiting in order to avoid merge conflicts (in the case that someone used this version in their refactor).

Reply:
It looks like you still use these methods in your branch though

Reply:
These are the legacy methods unrelated to this branch. I will remove them on a different User Story. 
they are still used in other students branches (login and editPost for example). I did not add these methods. I just changed the import to be explicit so that they would not be accidentally mistaken for the new User access methods. Removing these will be done in https://tree.taiga.io/project/dzemlin-poste/task/145

Reply:
Understood. Just want to make sure we are each incrementally moving our own tasks over to the new style so we can eventually remove, and confusion is mitigated.